guides/source/active_storage_overview.md

```diff
@@ -548,6 +548,57 @@ jobs, Cronjobs, etc.), you can access the `rails_blob_path` like this:
 Rails.application.routes.url_helpers.rails_blob_path(user.avatar, only_path: true)
 ```
 
+### Proxy mode
+
+Optionally, files can be proxied instead. This means that your application servers will download file data from the storage service in response to requests. This can be useful for serving files from a CDN.
+
+Explicitly proxy attachments using the `rails_storage_proxy_path` and `_url` route helpers:
+
+```erb
+<%= image_tag rails_storage_proxy_path(@user.avatar) %>
+```
+
+Or configure Active Storage to use proxying by default:
+
+```ruby
+# config/initializers/active_storage.rb
+Rails.application.config.active_storage.resolve_model_to_route = :rails_storage_proxy
+```
+
+In order to use a CDN for Active Storage attachments, you will need to generate URLs with proxy mode so that they are served by your app with the correct header (set by `http_cache_forever` method) so that any CDN will cache the attachment without any extra configuration. You should also make sure that the generated URLs use the CDN host instead of your app host. There are multiple ways to achieve this, but in general it involves tweaking your `config/routes.rb` file so that you can generate the proper URLs for the attachments and their variations. As an example you could add this:
```

Comment:
I might be missing something but it doesn't seem we're setting any additional headers w/rt/ `http_cache_forever` in the examples here.

Reply:
Also to re-quote what Andrew wrote here for context:

> can you point out the simplest solution is to put a CDN in front of your whole application - the default Active Storage proxy controller uses http_cache_forever which CDNs will cache by default without any extra configuration.

Reply:
>I might be missing something but it doesn't seem we're setting any additional headers w/rt/ http_cache_forever in the examples here.

That method is used here https://github.com/rails/rails/blob/c9614442b2102c9b8e828e9f6cbc9f91a62f6189/activestorage/app/controllers/active_storage/blobs/proxy_controller.rb#L8 so it's used by generating URLs with proxy mode. Does that answer it or did I misunderstand it?

>Also to re-quote what Andrew wrote here for context:

>>can you point out the simplest solution is to put a CDN in front of your whole application - the default Active Storage proxy controller uses http_cache_forever which CDNs will cache by default without any extra configuration.

We tried to point that out (maybe we didn't do a good job explaining it) and additionally explain give an example on how to generate URLs with the CDN host, which is IMHO the hardest part to get right given you need to deal with ActiveStorage internals as you can see in the example provided

Reply:
@santib In general, I think simple guides are great for starters but don't be afraid to provide more detailed examples as well that is the point of this guide. That said, maybe let's start small in this PR and open another for the nitty-gritty version.

Reply:
Great, we just pushed some tweaks hoping to make it more clear. Let me know your thoughts